src/manifest.c

```diff
@@ -877,6 +877,44 @@ void sort_manifest_by_version(struct manifest *manifest)
 	}
 }
 
+bool manifest_includes(struct manifest *manifest, char *component)
+{
+        GList *includes = g_list_first(manifest->includes);
+
+        while (includes) {
+                if (strcmp(((struct manifest *)includes->data)->component,
+                           component) == 0) {
+                        return true;
+                }
+                includes = g_list_next(includes);
+        }
+
+        return false;
+}
+
+bool changed_includes(struct manifest *old, struct manifest *new)
+{
+        GList *includes_old = g_list_first(old->includes);
+        GList *includes_new = g_list_first(new->includes);
+
+        if (g_list_length(includes_old) != g_list_length(includes_new)) {
+                return true;
+        }
+
+        while (includes_old && includes_new) {
+                char *bundle_old = ((struct manifest *)includes_old->data)->component;
+                char *bundle_new = ((struct manifest *)includes_new->data)->component;
+                if (strcmp(bundle_old, bundle_new) != 0) {
+                        return true;
+                }
```

Comment:
Should we assume the includes lists are sorted before calling this function? Comparing the bundle names like this assumes that.